File: pyg_seller_interface/src/main/java/com/songxin/pyg/seller/service/TypeTemplateService.java
package com.songxin.pyg.seller.service;
import java.util.List;
import com.songxin.pyg.pojo.TbTypeTemplate;

import com.songxin.pyg.vo.PageResultVO;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbTypeTemplate> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResultVO findByPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbTypeTemplate typeTemplate);
	
	
	/**
	 * 修改
	 */
	public void update(TbTypeTemplate typeTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbTypeTemplate findOneById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void batchDelete(Long[] ids);

	/**
	 * 根据条件分页查询
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResultVO findByCondition(TbTypeTemplate typeTemplate, int pageNum, int pageSize);
	
}

File: pyg_shop_web/src/main/webapp/js/service/typeTemplateService.js
//服务层
app.service("typeTemplateService", function ($http) {

	//读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../typeTemplate/findAll.do');
    };
    //分页查询所有
    this.findByPage = function (pageNum,pageSize) {
        return $http.get('../typeTemplate/findByPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize);
    };
    //根据条件分页查询
    this.findByCondition = function (pageNum, pageSize, conditionObj) {
        return $http.post("../typeTemplate/findByCondition.do?pageNum=" + pageNum + "&pageSize=" + pageSize, conditionObj);
    };
    //添加
    this.add = function (obj) {
        return $http.post('../typeTemplate/add.do', obj);
    };
    //根据id查找
    this.findOneById = function (id) {
        return $http.get("../typeTemplate/findOneById.do?id=" + id);
    };
    //修改
    this.update = function (obj) {
        return $http.post("../typeTemplate/update.do", obj);
    };
    //批量删除
    this.batchDelete = function (idsArr) {
        return $http.post("../typeTemplate/batchDelete.do", idsArr);
    };


});
